node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -361,71 +362,98 @@ pub mod standard {
-                        .clone()
+                    let err = err
                         .into_iter()
-                        .flat_map(|result| match result {
-                            Err(e) => Some(e),
-                            Ok(_) => None,
-                        })
-                        .collect::<Vec<ParamError>>();
-                    if errors.is_empty() {
-                        Ok(Some(
-                            results
-                                .into_iter()
-                                .filter(|result| result.is_ok())
-                                .map(|result| result.expect("Error materialized"))
-                                .collect::<Vec<NodeDescriptor>>(),
-                        ))
+                        .map(|err| err.expect_err("ParamError"))
+                        .collect_vec();
+                    if err.is_empty() {
+                        Ok(Some(ok))
                     } else {
-                        Err(ConfiguratorError::new(errors))
+                        Err(ConfiguratorError::new(err))
                     }
                 }
             }
         }
     }
 
+    fn validate_descriptors_from_user(
+        descriptors: Vec<String>,
+        dummy_cryptde: Box<dyn CryptDE>,
+        desired_chain: Chain,
+    ) -> Vec<Result<NodeDescriptor, ParamError>> {
+        descriptors.into_iter()
+            .map(
+                |node_desc_from_ci| {
+                    let node_desc_trimmed = node_desc_from_ci.trim();
+                    match NodeDescriptor::try_from((dummy_cryptde.as_ref(), node_desc_trimmed)) {
+                        Ok(descriptor) =>
+                            {
+                                let competence_from_descriptor = descriptor.blockchain;
+                                if desired_chain == competence_from_descriptor {
+                                    validate_mandatory_node_addr(node_desc_trimmed, descriptor)
+                                } else {
+                                    let name_of_desired_chain = desired_chain.rec().literal_identifier;
+                                    Err(ParamError::new("neighbors", &format!("Mismatched chains. You are requiring access to '{}' ({}{}:<public key>@<node address>) with descriptor belonging to '{}'",
+                                                                              name_of_desired_chain, MASQ_URL_PREFIX,
+                                                                              name_of_desired_chain,
+                                                                              competence_from_descriptor.rec().literal_identifier)))
+                                }
+                            }
+                        Err(e) => ParamError::new("neighbors", &e).wrap_to_err()
+                    }
+                }
+            )
+            .collect_vec()
+    }
+
+    fn validate_mandatory_node_addr(
+        supplied_version: &str,
```

Comment:
Don't say "version;" it makes the reader think of "1.2.3" or something. Maybe user_supplied_descriptor? Or just supplied_descriptor, since it might come from the environment or the config file as well as the command line?